File: Java/1108. Defanging an IP Address.java
/**
 * https://leetcode.com/problems/defanging-an-ip-address/
 */

public class DefangingAnIPAddress {
    // O(n) time complexity
    // O(n) space complexity
    String defangIPaddr(String address) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < address.length(); i++) {
            char c = address.charAt(i);
            if (Character.isDigit(c)) {
                sb.append(c - '0');
            } else {
                sb.append("[.]");
            }
        }
        return sb.toString();
    }
}

class T {
    public static void main(String[] args) {
        DefangingAnIPAddress d = new DefangingAnIPAddress();
        String input = "255.100.50.0";
        System.out.println(d.defangIPaddr(input)); // Output: 255[.]100[.]50[.]0
    }
}

File: Kotlin/1108. Defanging an IP Address.kt
/**
 * https://leetcode.com/problems/defanging-an-ip-address/
 */

// O(n) time complexity
// O(n) space complexity
fun defangIPaddr(address: String): String {
    val sb = StringBuilder()
    address.forEach { c ->
        if (c.isDigit()) sb.append(c)
        else sb.append("[.]")
    }
    return sb.toString()
}

fun main() {
    val input = "255.100.50.0"
    println(defangIPaddr(input)) // Output: 255[.]100[.]50[.]0

}